src/main/kotlin/org/elm/workspace/compiler/ElmBuildAction.kt

```diff
@@ -0,0 +1,81 @@
+package org.elm.workspace.compiler
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.util.messages.Topic
+import org.elm.lang.core.lookup.ClientLocation
+import org.elm.lang.core.lookup.ElmLookup
+import org.elm.lang.core.psi.ElmNamedElement
+import org.elm.lang.core.types.TyUnion
+import org.elm.lang.core.types.findInference
+import org.elm.openapiext.saveAllDocuments
+import org.elm.workspace.ElmProject
+import org.elm.workspace.elmToolchain
+import org.elm.workspace.elmWorkspace
+
+class ElmBuildAction : AnAction() {
+
+    private val elmMainTypes: Set<Pair<String, String>> = hashSetOf(Pair("Platform", "Program"), Pair("Html", "Html"))
+
+    private val elmJsonReport = ElmJsonReport()
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project
+                ?: return
+        saveAllDocuments()
+
+        val elmCLI = project.elmToolchain.elmCLI
+        if (elmCLI == null) {
+            Messages.showErrorDialog("No path to the Elm compiler", "Build Error")
+            return
+        }
+
+        val elmProject = project.elmWorkspace.allProjects.first()
+
+        // find "main" function
+        val clientLocation = LookupClientLocation(project, elmProject)
+        val elements = ElmLookup.findByName<ElmNamedElement>("main", clientLocation)
```

Comment:
rather than restricting the return type to `ElmNamedElement`, instead use `ElmFunctionDeclarationLeft`